### PR TITLE
Move vk_safe_struct.h to VUL

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
This code was being generated in both Vulkan-ValidationLayers and Vulkan-ExtensionLayer. Further uses are on the horizon so lets stop the copypasta.

### CODE REVIEW

**Comment by jeremyg-lunarg** [comment]
This is draft until I prove it works for VUL and VEL

File: include/vulkan/utility/concurrent_unordered_map.h
Comment (jeremyg-lunarg) [inline include/vulkan/utility/concurrent_unordered_map.h:198]: This TODO is where we'd want to use  the robin hood or unordered_dense hashmaps via the vvl::unordered_map alias in the original code.

File: scripts/common_ci.py
Comment (jeremyg-lunarg) [inline scripts/common_ci.py:1]: I pulled over lots of python changes to make the safe struct generators and clang-format work. It is ugly copy copy though.

**Review by jeremyg-lunarg — COMMENTED** [review]
This would be a good opportunity to change the safe names from safe_VkInstanceCreateInfo to something like `vku::safe::InstanceCreateInfo` if we want to do that.

File: src/vulkan/CMakeLists.txt
Comment (jeremyg-lunarg) [reply src/vulkan/CMakeLists.txt:25]: pick which confusion you prefer. :)

File: scripts/generators/generator_utils.py
Comment (spencer-lunarg) [inline scripts/generators/generator_utils.py:91]: not needed

File: scripts/generators/vulkan_object.py
Comment (spencer-lunarg) [inline scripts/generators/vulkan_object.py:1]: so sorry I haven't got to a point where this is all unified in put the `Vulkan-Headers` yet, but we can have a seperate PR where we unify `vulkan_object.py` and `base_generator.py` across everywhere

File: scripts/generators/generator_utils.py
Comment (jeremyg-lunarg) [reply scripts/generators/generator_utils.py:91]: I'm trying to keep the customization of scripts to a minimum and just using the VVL versions as a the known good version

**Comment by jeremyg-lunarg** [comment]
> This would be a good opportunity to change the safe names from safe_VkInstanceCreateInfo to something like `vku::safe::InstanceCreateInfo` if we want to do that.

Changing to this naming convention is problematic because safe_VkMemoryBarrier becomes vku::safe::MemoryBarrier. Windows has a macro named MemoryBarrier which then breaks all over the place. Both this code and client code would need to #undef that macro.

File: scripts/generators/generator_utils.py
Comment (charles-lunarg) [inline scripts/generators/generator_utils.py:77]:
```suggestion
        if self.current_guard != guard and self.current_guard is not None:
            out.append(f'#endif  // {self.current_guard}\n')
        if extra_newline:
            out.append('\n')
        if self.current_guard != guard and guard is not None:
            out.append(f'#ifdef {guard}\n')
        self.current_guard = guard
```
Apparently replacing `!=` with `is not` provokes different behavior.

File: include/vulkan/vk_enum_string_helper.h
Comment (charles-lunarg) [inline include/vulkan/vk_enum_string_helper.h:919]:
```suggestion
        case VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SHADER_ENQUEUE_PROPERTIES_AMDX:
            return "VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SHADER_ENQUEUE_PROPERTIES_AMDX";
        case VK_STRUCTURE_TYPE_EXECUTION_GRAPH_PIPELINE_SCRATCH_SIZE_AMDX:
            return "VK_STRUCTURE_TYPE_EXECUTION_GRAPH_PIPELINE_SCRATCH_SIZE_AMDX";
        case VK_STRUCTURE_TYPE_EXECUTION_GRAPH_PIPELINE_CREATE_INFO_AMDX:
            return "VK_STRUCTURE_TYPE_EXECUTION_GRAPH_PIPELINE_CREATE_INFO_AMDX";
```

Comment (charles-lunarg) [reply include/vulkan/vk_enum_string_helper.h:919]: This is what ticked me off to the guard_helper regressing.

**Review by charles-lunarg — CHANGES_REQUESTED** [review]

File: scripts/generators/generator_utils.py
Comment (jeremyg-lunarg) [reply scripts/generators/generator_utils.py:77]: ok. that means VVL is broken too, because I just copied the 'newer' scripts over... We have script madness.

Comment (jeremyg-lunarg) [reply scripts/generators/generator_utils.py]: should be fixed now. I'll push a fix to VVL in a sec

Comment (jeremyg-lunarg) [reply scripts/generators/generator_utils.py]: https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/7787

File: src/CMakeLists.txt
Comment (charles-lunarg) [inline src/CMakeLists.txt:31]: Normally I'd be hesitant to add warnings as this target is intended to be consumed downstream. HOWEVER 1. I don't think anyone uses it and 2. this w4324 seems to be a real nonsense warning due to alignment.

**Review by charles-lunarg — APPROVED** [review]
Most of the changes are not code I really *should* review. As a lot of this code comes straight from VVL. But I did go over the bits that aren't in VVL well enough to feel confident.

Of course, it would be great if @jeremyg-lunarg used this PR in VVL before merging it to make sure it 'works' but safe structs is SO HUGE that I don't see a reason to hold up this PR while VVL is moved over. If fixes are needed to VUL's safe structs, that is just gonna have to happen.

**Comment by jeremyg-lunarg** [comment]
> Of course, it would be great if @jeremyg-lunarg used this PR in VVL before merging

There are draft PRs for VEL (https://github.com/KhronosGroup/Vulkan-ExtensionLayer/pull/378)and VVL (https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/7717)  that use this PR.

File: src/CMakeLists.txt
Comment (jeremyg-lunarg) [reply src/CMakeLists.txt:31]: this warning comes from vku::concurrent::unordered_map which purposefully adds padding for better multithreaded performance.

**Pull request merged**
